### PR TITLE
 JAMES-2167 Serializable attributes on a JMS queue [CONTRIB]

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
CI for https://github.com/apache/james-project/pull/126

**Comment by chibenwa** [comment]
```
[e0ea49926d0ee3685f82677c85d4fb8565137553] Tests in error: 
[e0ea49926d0ee3685f82677c85d4fb8565137553]   ForwardSmtpTest.forwardingAnEmailShouldWork:81 » ConditionTimeout Condition wi...
```

test this please

File: server/queue/queue-api/src/test/java/org/apache/james/queue/api/MailQueueContract.java [hunk 1]
```diff
@@ -305,4 +322,33 @@ default void deQueueShouldWaitForAMailToBeEnqueued(ExecutorService executorServi
...
+        private final String value;
+
+        public SerializableAttribute(String value) {
+            this.value = value;
+        }
+
+        @Override
+        public final boolean equals(Object o) {
+            if (o instanceof SerializableAttribute) {
+                SerializableAttribute toto = (SerializableAttribute) o;
```
Comment (aduprat) [inline server/queue/queue-api/src/test/java/org/apache/james/queue/api/MailQueueContract.java]: s/toto/that

File: server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSSerializationUtils.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+     * Decodes the input base64 string and deserialize it.
+     *
+     * @param <T>    The resulting type after deserialization.
+     * @param object The base64 encoded string.
+     *
+     * @return The deserialized object.
+     */
+    public static <T extends Serializable> T deserialize(String object) {
+        return Optional.ofNullable(object)
+                .map(String::getBytes)
```
Comment (rouazana) [inline server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSSerializationUtils.java]: missing encoding

File: server/queue/queue-jms/src/test/java/org/apache/james/queue/jms/JMSSerializationUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    @Test
+    void deserializeShouldThrowWhenNotBase64StringProvided() {
+        assertThatExceptionOfType(SerializationException.class)
+                .isThrownBy(() -> deserialize("abc"));
+    }
+
+    @Test
+    void deserializeShouldThrowWhenNotSerializedBytesAreEncodedInBase64() {
+        assertThatExceptionOfType(SerializationException.class)
+                .isThrownBy(() -> deserialize(Base64.encodeBase64String("abc".getBytes())));
```
Comment (rouazana) [inline server/queue/queue-jms/src/test/java/org/apache/james/queue/jms/JMSSerializationUtilsTest.java]: missing encoding.
And I don't understand what this test is doing

Comment (chibenwa) [reply server/queue/queue-jms/src/test/java/org/apache/james/queue/jms/JMSSerializationUtilsTest.java]: It's base 64 but it's not serialized hence deserializing it should fail. (edge case testing)

**Comment by chibenwa** [comment]
Merged
